src/shogun/lib/SGMatrix.h

```diff
@@ -306,9 +309,60 @@ template<class T> class SGMatrix : public SGReferencedData
 		/** Set matrix to a constant */
 		void set_const(T const_elem);
 
-		/** fill matrix with zeros */
+		/** Fill matrix with zeros */
 		void zero();
 
+		/**
+		 * Fill matrix with uniformly distributed randoms
+		 * Float Types: [0, 1)
+		 * Int Types: std::numeric_limit<T>::(min->max)
+		**/
+		template <typename U>
+		using enable_simd_float = std::enable_if_t<sg_is_same_v<U, float64_t>>;
+		template <typename U = T>
+		auto random() -> enable_simd_float<U>
+		{
+			auto r = std::make_unique<CRandom>();
```

Comment:
the problem with having CRandom created here is that you wont ever be able to set seed :(
there's a way to do this, check the usage of the global variable `sg_rand` in the codebase... but i rather would start reviving the `feature/random-refactor` branch and have then this fixed....

Reply:
the rest of the patch has the same pattern and it'll be a problem as well there...

Reply:
ive addressed it for `SGMatrix`
`SGVector`/`CRandom` are being included in one another, which is causing a problem when using `sg_rand` in the header directly. I'll have a go at it tomorrow.

Reply:
done

Reply:
nice!